sirepo/job_supervisor.py

```diff
@@ -300,6 +300,11 @@ def _purge_sim(db_file):
             cls._purged_jids_cache.add(db_file.purebasename)
             if d.status == job.JOB_RUN_PURGED:
                 return
+            # compute_model isn't guaranteed to be in data but simulation_db.simulation_run_dir()
+            # expects it
+            d.pksetdefault(
+                computeModel=sirepo.sim_data.split_jid(db_file.purebasename).compute_model
```

Comment:
@robnagler maybe computeModel should be a required field and we should create a fixup in __db_load() for it? Also, we should probably use __db_load here so we get the fixups it applies.

Reply:
When would it not be in data?

Reply:
We don't require it in [__db_init_new()](https://github.com/radiasoft/sirepo/blob/c1f31e8b286fb373dfcf1779c12316a931084ae2/sirepo/job_supervisor.py#L390). It is only put on in [runSimulaiton()](https://github.com/radiasoft/sirepo/blob/c1f31e8b286fb373dfcf1779c12316a931084ae2/sirepo/job_supervisor.py#L689). In addition our upgrade code [doesn't put it on](https://github.com/radiasoft/sirepo/blob/c1f31e8b286fb373dfcf1779c12316a931084ae2/sirepo/pkcli/db.py#L84). I think it should be a required field.

I've fixed this in  https://github.com/radiasoft/sirepo/pull/2534/commits/788682ba9f66212284a3762464e80326e208fc36. Our db upgrade code now includes it. In addition, __db_load() fixes up the data and puts computeModel on. Our purge code now uses __db_load() so it will get all fixups.